### DISCUSSION TITLE
Setting Cache-Control headers with stale-while-revalidate

### DISCUSSION THREAD

**Discussion by olololushky** [vercel/community#1297] [Help] [answered]
### Summary

Hi there!
I'm trying to set up Edge function with caching responses in Vercel CDN with revalidation response in background
But actually, it always returns original response, even if it already marked as stale, and actual response should be different
As I can see in function logs, even if header x-vercel-cache shows STALE, nothing happens
Is there should be any specific setup except setting up headers into the Edge function response? I tried different option with client cache control headers, and an adding Tag into the Edge function response, but it also doen't work for me

I'm testing with this params, but will increase after going live. 
"Cache-Control": "max-age=0, s-maxage=60, stale-while-revalidate=120"
Thanks everyone!

**Comment by amyegan** [comment]
This is a great question @olololushky. This was also brought up in issue https://github.com/vercel/vercel/issues/9189 and our support team is looking into it to get the full picture of what is happening. 🙇 

In general, I recommend that everyone read through the guide about [how to configure the Cache-Control response header in Vercel Projects](https://vercel.com/guides/how-to-configure-the-cache-control-response-header-in-vercel-projects) for help with the general setup. But that doesn't answer the question about Edge Functions specifically. Stay tuned because we'll share updates as we get more info from the engineering team.

**Reply by smaeda-ks** [reply]
Hi @olololushky, we've just shipped a fix for this bug. It should be working now. Could you please confirm that? Thanks!

**Reply by olololushky** [reply, marked as answer]
Hi!
Now it works for me
Thank you!
